src/evict/evict_page.c

```diff
@@ -676,7 +676,6 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     /*
      * Success: assert that the page is clean or reconciliation was configured to save updates.
      */
-    WT_ASSERT(session, !__wt_page_is_modified(page) || LF_ISSET(WT_REC_HS | WT_REC_IN_MEMORY));
```

Comment:
I don't understand why this is removed.

Reply:
Well eviction can now evict a history store page and it will no longer be clean with the new conditional.